iOS/IssueTracker/IssueTracker/Issue/IssueCollectionViewDataSource.swift

```diff
@@ -9,17 +9,47 @@ import UIKit
 
 class IssueCollectionViewDataSource: NSObject, UICollectionViewDataSource {
     
-    private var issueData: [IssueItem] = [.init(title: "제목", issueDescription: "이슈에 대한 설명(최대 두 줄까지 보여줄 수 있다)", milestoneName: "마일스톤 이름") ]
+    var data: [IssueItem] = [
+        .init(id: 1, title: "제목", content: "이슈에 대한 설명", milestoneName: "마일스톤 이름", labels: [
+            .init(title: "레이블 이름", backgroundColor: "#677ae5")
+        ]),
+        .init(id: 2, title: "제목", content: "이슈에 대한 설명", milestoneName: "마일스톤 이름", labels: [
+            .init(title: "Bug", backgroundColor: "#20bb87"),
+            .init(title: "iOS", backgroundColor: "#20bb87")
+        ])
+    ]
+    
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
     
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        issueData.count
+        data.count
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: IssueCollectionViewCell.identifier, for: indexPath) as? IssueCollectionViewCell else {
             return UICollectionViewCell()
         }
         
+        cell.configure(title: data[indexPath.item].title,
+                       issueDescription: data[indexPath.item].content,
+                       milestoneName: data[indexPath.item].milestoneName,
+                       labelName: data[indexPath.item].labels[0].title,
+                       labelBackgroundColor: data[indexPath.item].labels[0].backgroundColor)
+        
         return cell
     }
+    
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        if kind == UICollectionView.elementKindSectionHeader {
+            guard let header = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: IssueCollectionHeaderView.identifier, for: indexPath) as? IssueCollectionHeaderView else {
+                return UICollectionReusableView()
+            }
+            header.setHeaderTitle(text: "이슈")
+            return header
+        } else {
+            return UICollectionReusableView()
+        }
```

Comment:
객체지향을 제대로 연습해보고 싶다면 이런 상황에서 if-else를 사용하지 않는 연습을 해보는 것도 좋은거 같아요!

Reply:
최대한 Case별로 나누어 보면 좋겠다는 의견으로 생각이 들어 아래와 같이 변경해보았습니다!
```Swift
switch kind {
        case UICollectionView.elementKindSectionHeader:
            guard let header = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: IssueCollectionHeaderView.identifier, for: indexPath) as? IssueCollectionHeaderView else {
                return UICollectionReusableView()
            }
            let sectionType = SectionList.allCases[indexPath.section]
            switch sectionType {
            case .issue:
                header.setHeaderTitle(text: "이슈")
            case .otherwise:
                header.setHeaderTitle(text: "")
        }
            return header
        default:
            return UICollectionReusableView()
        }
    }
    
    enum SectionList: Int, CaseIterable {
        case issue = 0
        case otherwise
    }
```

Reply:
SecionList가 엄청 커져서 8개 10개 이상이 된다면 관리하기가 쉽지 않아질거 같은데요! 이런 부분을 다형성을 통해서 리팩터링 해볼 수 있다면 굉장히 좋을거 같습니다....!! 👍